app/layersmodel.cpp

```diff
@@ -13,126 +7,35 @@
  *                                                                         *
  ***************************************************************************/
 
-#include "layersmodel.h"
-#include "qgsquickmapsettings.h"
-
-#include <qgslayertreemodel.h>
-#include <qgslayertreenode.h>
-#include <qgslayertree.h>
-#include <qgsvectorlayer.h>
-#include <qgslayertreemodellegendnode.h>
-#include <qgsproject.h>
-#include "qgsvectorlayer.h"
-#include <qgsmaplayer.h>
-#include <qgswkbtypes.h>
-
-#include <QString>
-
-LayersModel::LayersModel( QObject *parent )
-  : QAbstractListModel( parent )
-{
-}
-
-LayersModel::~LayersModel()
-{
-}
-
-
-void LayersModel::reloadLayers( QgsProject *project )
-{
-
-  if ( !project ) return;
-
-  QgsLayerTreeGroup *root = project->layerTreeRoot();
-
-  // Get list of all visible and valid layers in the project
-  QList< QgsMapLayer * > allLayers;
-  foreach ( QgsLayerTreeLayer *nodeLayer, root->findLayers() )
-  {
-    if ( nodeLayer->isVisible() )
-    {
-      QgsMapLayer *layer = nodeLayer->layer();
-      if ( layer->isValid() )
-      {
-        allLayers << layer;
-        qDebug() << "Found layer: " << layer->name();
-      }
-    }
-  }
-  if ( mLayers != allLayers )
-  {
-    beginResetModel();
-    setLayers( allLayers );
-    endResetModel();
-  }
-}
 
-QgsQuickMapSettings *LayersModel::mapSettings() const
-{
-  return mMapSettings;
-}
+#include "layersmodel.h"
 
-void LayersModel::setMapSettings( QgsQuickMapSettings *mapSettings )
+LayersModel::LayersModel()
 {
-  if ( mapSettings != mMapSettings )
-  {
-    mMapSettings = mapSettings;
-    setLayers( mLayers );
-    emit mapSettingsChanged();
-  }
 }
 
-void LayersModel::updateActiveLayer( const QString &name )
+QVariant LayersModel::data( const QModelIndex &index, int role ) const
 {
-  int row = rowAccordingName( name, firstWritableLayerIndex() );
-  setActiveIndex( row );
-}
+  if ( !index.isValid() )
+    return QVariant();
 
-int LayersModel::activeIndex() const
-{
-  return mActiveIndex;
-}
+  if ( role < LayerNameRole ) // if requested role from parent
+    return QgsMapLayerModel::data( index, role );
 
-QString LayersModel::setActiveIndex( int activeIndex )
-{
-  if ( mActiveIndex != activeIndex )
-  {
-    mActiveIndex = activeIndex;
-  }
-  emit activeIndexChanged();
-
-  QgsMapLayer *layer = activeLayer();
-  if ( layer )
-    return layer->name();
-  else
-    return QString();
-}
+  QgsVectorLayer *vectorLayer;
+  QgsMapLayer *layer;
```

Comment:
IMHO it's better to declare local variables at the same time they are first initialized... this way I need to check further down to make sure they don't stay uninitialized and it's two lines of code more :-)